lib/msf/core/exploit/smb/client.rb

```diff
@@ -0,0 +1,636 @@
+          end
+        end
+
+        sp
+      end
+
+
+      #
+      # Determine the native language pack of a Windows system via SMB probes
+      #
+      def smb_fingerprint_windows_lang
+
+        #
+        # Remote language detection via Print Providers
+        # Credit: http://immunityinc.com/downloads/Remote_Language_Detection_in_Immunity_CANVAS.odt
+        #
+
+        lang = 'Unknown'
+
+        sigs =
+        {
+          'English' =>
+            [
+              Rex::Text.to_unicode('Windows NT Remote Printers'),
+              Rex::Text.to_unicode('LanMan Print Services')
+            ],
+          'Spanish' =>
+            [
+              Rex::Text.to_unicode('Impresoras remotas Windows NT'),
+              Rex::Text.to_unicode('Impresoras remotas de Windows NT')
+            ],
+          'Italian' =>
+            [
+              Rex::Text.to_unicode('Stampanti remote di Windows NT'),
+              Rex::Text.to_unicode('Servizi di stampa LanMan')
+            ],
+          'French' =>
+            [
+              Rex::Text.to_unicode('Imprimantes distantes NT'),
+              Rex::Text.to_unicode('Imprimantes distantes pour Windows NT'),
+              Rex::Text.to_unicode("Services d'impression LanMan")
+            ],
+          'German' =>
+            [
+              Rex::Text.to_unicode('Remotedrucker')
+            ],
+          'Portuguese - Brazilian' =>
+            [
+              Rex::Text.to_unicode('Impr. remotas Windows NT'),
+              Rex::Text.to_unicode('Impressoras remotas do Windows NT')
+            ],
+          'Portuguese' =>
+            [
+              Rex::Text.to_unicode('Imp. remotas do Windows NT')
+            ],
+          'Hungarian' =>
+            [
+              Rex::Text.to_unicode("\x54\xe1\x76\x6f\x6c\x69\x20\x6e\x79\x6f\x6d\x74\x61\x74\xf3\x6b")
+            ],
+          'Finnish' =>
+            [
+              Rex::Text.to_unicode("\x45\x74\xe4\x74\x75\x6c\x6f\x73\x74\x69\x6d\x65\x74")
+            ],
+          'Dutch' =>
+            [
+              Rex::Text.to_unicode('Externe printers voor NT')
+            ],
+          'Danish' =>
+            [
+              Rex::Text.to_unicode('Fjernprintere')
+            ],
+          'Swedish' =>
+            [
+              Rex::Text.to_unicode("\x46\x6a\xe4\x72\x72\x73\x6b\x72\x69\x76\x61\x72\x65")
+            ],
+          'Polish' =>
+            [
+              Rex::Text.to_unicode('Zdalne drukarki')
+            ],
+          'Czech'   =>
+            [
+              Rex::Text.to_unicode("\x56\x7a\x64\xe1\x6c\x65\x6e\xe9\x20\x74\x69\x73\x6b\xe1\x72\x6e\x79")
+            ],
+          'Turkish' =>
+            [
+              "\x59\x00\x61\x00\x7a\x00\x31\x01\x63\x00\x31\x01\x6c\x00\x61\x00\x72\x00"
+            ],
+          'Japanese' =>
+            [
+              "\xea\x30\xe2\x30\xfc\x30\xc8\x30\x20\x00\xd7\x30\xea\x30\xf3\x30\xbf\x30"
+            ],
+          'Chinese - Traditional' =>
```

Comment:
Where is Simplified Chinese?

And what's the difference between this "Traditional Chinese" vs the one in Taiwan?